### PR TITLE
TASK-51257 :  Redirect space wallet setting to the space's settings

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue [hunk 1]
```diff
@@ -49,14 +50,21 @@ export default {
...
         const type = this.getQueryParam('type');
         this.getWallet(id,type);
         window.history.replaceState('wallet', 'My wallet', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/settings`);
         document.location.hash = '#walletSettings';
       }
+      if (this.from === 'space') {
+        const id = this.getQueryParam('id');
+        const type = this.getQueryParam('type');
+        this.getWallet(id,type);
+        window.history.replaceState('wallet', 'My wallet', `${eXo.env.portal.context}/g/:spaces:${eXo.env.portal.spaceGroup}/${eXo.env.portal.spaceName}/settings/settings`);
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: the string 'My Wallet' is not correct IMO, and should be replaced with the value of an internationalized one.

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue [hunk 2]
```diff
@@ -71,7 +79,11 @@ export default {
     closeDetail() {
       document.dispatchEvent(new CustomEvent('showSettingsApps'));
       this.displayDetails = false;
-      window.history.replaceState('wallet', 'My wallet', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/settings`);
+      if (this.from === 'space'){
+        window.history.replaceState('wallet', 'My wallet', `${eXo.env.portal.context}/g/:spaces:${eXo.env.portal.spaceGroup}/${eXo.env.portal.spaceName}/settings/settings`);
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: th esame as above

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue (hunk 1 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: done!

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue (hunk 2 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: done!

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue [hunk 3]
```diff
@@ -63,15 +78,24 @@ export default {
...
       document.dispatchEvent(new CustomEvent('hideSettingsApps', {detail: this.id}));
       this.getWallet();
       this.displayDetails = true;
+      this.init();
     },
     closeDetail() {
       document.dispatchEvent(new CustomEvent('showSettingsApps'));
       this.displayDetails = false;
-      window.history.replaceState('wallet', 'My wallet', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/settings`);
+      if (this.from === 'space' || eXo.env.portal.spaceName !== ''){
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]:
```suggestion
      if (this.from === 'space' || eXo.env.portal.spaceName){
```

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue [hunk 4]
```diff
@@ -63,15 +78,24 @@ export default {
...
+        window.history.replaceState('wallet', this.$t('exoplatform.wallet.title.myWallet'), `${eXo.env.portal.context}/g/:spaces:${eXo.env.portal.spaceGroup}/${eXo.env.portal.spaceName}/settings`);
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: I believe we need another resource bundle key for the page title.

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue [hunk 5]
```diff
@@ -80,8 +104,8 @@ export default {
     },
     getWallet(id,type) {
       if (!id){
-        id = eXo.env.portal.userName ;
-        type ='user';
+        id = eXo.env.portal.spaceName !== '' ? eXo.env.portal.spaceName : eXo.env.portal.userName ;
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]:
```suggestion
        id = eXo.env.portal.spaceName ? eXo.env.portal.spaceName : eXo.env.portal.userName ;
```

**Review by ahamdi — APPROVED** [review]

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue (hunk 3 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: done!

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue (hunk 4 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: done!

File: wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue (hunk 5 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/vue-app/wallet-settings/components/WalletSettings.vue]: done!

File: wallet-webapps-common/src/main/webapp/css/main.less [hunk 6]
```diff
@@ -34,6 +34,14 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
 }
 
+#SpacePage {
+  .walletSettingsCard {
+    margin: 0px !important;
+  }
+  .walletSetting {
+    margin: 0px !important;
+  }
+}
```
Comment (ahamdi) [inline wallet-webapps-common/src/main/webapp/css/main.less]: could you use CSS helpers for this ?

File: wallet-webapps/src/main/webapp/vue-app/initComponents.js [hunk 7]
```diff
@@ -18,6 +19,13 @@ const components = {
   'wallet-reward-transaction-history-chart-summary': TransactionHistoryChartSummary,
 };
 
+const externalComponentOptions = {
+  name: 'space-wallet-setting',
+  componentImpl: Vue.options.components['wallet-settings']
+};
+
+document.dispatchEvent(new CustomEvent('addExternalComponents', {'detail': externalComponentOptions}));
```
Comment (ahamdi) [inline wallet-webapps/src/main/webapp/vue-app/initComponents.js]: Could you use a more precise name for the event ? addSpaceSettingsComponents

**Review by ahamdi — DISMISSED** [review]

File: wallet-webapps-common/src/main/webapp/css/main.less (hunk 6 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/css/main.less]: we used the ma-4 class in the walletSttings in the userSetting which is not practical in the spaceSetting so we added this!

Comment (ahamdi) [reply wallet-webapps-common/src/main/webapp/css/main.less]: Could it be done with a computed property ?

File: wallet-webapps/src/main/webapp/vue-app/initComponents.js (hunk 7 above)
Comment (GouadriaHanen) [reply wallet-webapps/src/main/webapp/vue-app/initComponents.js]: done!

File: wallet-webapps-common/src/main/webapp/css/main.less (hunk 6 above)
Comment (GouadriaHanen) [reply wallet-webapps-common/src/main/webapp/css/main.less]: done!

**Review by Rdidmohamedelhadi — APPROVED** [review]
LGTM

**Pull request merged**